csharp/src/Google.Protobuf/ParseContext.cs

```diff
@@ -74,6 +75,7 @@ internal static void Initialize(ReadOnlySpan<byte> buffer, out ParseContext ctx)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         internal static void Initialize(ReadOnlySpan<byte> buffer, ref ParserInternalState state, out ParseContext ctx)
```

Comment:
I guess we are not using this anywhere else? Else we'd likely get the same build errors elsewhere? Can't you just remove this internal method?

Reply:
Ah yes I see, because the `state` used there is not a local variable in a method, it's and instance variable on the CodedInputStream class so it's already on the heap, not on the stack (the problem seems to be because of leaking a ref to a by value type variable that's local to a method and thus allocated in the stack and not the heap, as I understand it).
Passing the `state` by ref here is just for optimizing, right, so that it's not copied? But I wonder, is that even necessary when the method is marked for aggresive inlining? Anyway, LGTMing, this is more to maybe avoid the nuisance of 
"chenge there if you change here" comment.